src/test/merkleblock_tests.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2012-2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+                          
+#include "merkleblock.h"
+#include "uint256.h"
+#include "test/test_bitcoin.h"
+ 
+#include <boost/test/unit_test.hpp>
+ 
+
+BOOST_FIXTURE_TEST_SUITE(merkleblock_tests, BasicTestingSetup)
+
+/**
+ * Create a CMerkleBlock using a list of txids which will be found in the 
+ * given block.
+ */
+BOOST_AUTO_TEST_CASE(merkleblock_construct_from_txids_found)
+{ 
+    CBlock block = getBlock13b8a();
+
+    std::set<uint256> txids1;
+    uint256 txhash1 = uint256S("0x74d681e0e03bafa802c8aa084379aa98d9fcd632ddc2ed9782b586ec87451f20");
+    txids1.insert(txhash1);
+    CMerkleBlock merkleBlock(block, txids1);
+
+    BOOST_CHECK_EQUAL(merkleBlock.header.GetHash().GetHex(), block.GetHash().GetHex());
+    BOOST_CHECK_EQUAL(merkleBlock.vMatchedTxn.size(), 0);
```

Comment:
Seems a strange thing to test, here? Maybe document on merkleblock.h what this thing even means? (ie "Set only when constructing from a bloom filter to allow testing the set of things which matched the bloom filter directly").

Reply:
Comment added inline with test as well as merkleblock.h. Basically just used your copy directly.